File: problems/Codeforces/746B/solution.cpp
#include <iostream>
#include <string>
#include <vector>

int main() {
    using namespace std;
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int _;
    string word;
    cin >> _ >> word;
    vector<char> res;
    for(int i = word.length() - 1, j=0 ; i >= 0 ; i--, j++) {
        auto pos = j%2==0 ? res.size() - j/2 : j/2;
        res.insert(res.begin() + pos, word[i]);
    }
    cout.write(res.data(), res.size());
    return 0;
}

File: problems/Codeforces/746B/solution.rs
use std::io::{self, Read, Write, BufWriter};

fn main() -> io::Result<()> {
    let mut input = String::new();
    io::stdin().lock().read_to_string(&mut input)?;
    let mut iter = input.split_ascii_whitespace();
    let mut writer = BufWriter::new(io::stdout().lock());

    iter.next();
    let mut res = Vec::new();
    for (i, &c) in iter.next().unwrap().as_bytes().to_vec().iter().rev().enumerate() {
        let pos = if i % 2 == 0 {
            res.len() - i / 2
        } else {
            i / 2
        };
        res.insert(pos, c);
    }
    writer.write_all(&mut res)?;
    Ok(())
}
